src/client/pmi1.c

```diff
@@ -59,20 +61,37 @@ static int pmi_init = 0;
 
 int PMI_Init(int *spawned)
 {
+    pmix_status_t rc = PMIX_SUCCESS;
     pmix_value_t *val;
-    pmix_status_t rc;
+    pmix_proc_t proc;
+    pmix_info_t info[2];
+    bool  val_optinal = 1;
+    int val_timeout = PMIX_TIMEOUT_VALUE;
 
     if (PMIX_SUCCESS != PMIx_Init(&myproc)) {
         return PMI_ERR_INIT;
     }
 
+    /* getting internal key requires special rank value */
+    memcpy(&proc, &myproc, sizeof(myproc));
+    proc.rank = PMIX_RANK_UNDEF;
+
+    /* set controlling parameters
+     * PMIX_OPTIONAL - expect that these keys should be available on startup
+     * PMIX_TIMEOUT - meet legacy PMI requirement
+     */
+    PMIX_INFO_CONSTRUCT(&info[0]);
+    PMIX_INFO_CONSTRUCT(&info[1]);
+    PMIX_INFO_LOAD(&info[0], PMIX_OPTIONAL, &val_optinal, PMIX_BOOL);
+    PMIX_INFO_LOAD(&info[1], PMIX_TIMEOUT, &val_timeout, PMIX_INT);
```

Comment:
I can see doing this for testing, but I really am not comfortable with hard-coding timeout limits in the PMIx library itself. The time required for an operation will depend on the size of the job, the transport being used, the RM's efficiency, etc. It isn't something we should hard-code.